strawberry/types/fields/resolver.py

```diff
@@ -171,6 +172,12 @@ def is_reserved_type(self, other: builtins.type) -> bool:
 
 T = TypeVar("T")
 
+#  asyncio.iscoroutinefunction was deprecated in python >= 3.12
```

Comment:
it is not that it was deprecated, but actually that python 3.12 changed the way to mark something as a coroutine by requiring it to be marked using [inspect.markcoroutinefunction](https://docs.python.org/3/library/inspect.html#inspect.markcoroutinefunction), and that needs to be checked with [inspect.iscoroutinefunction](https://docs.python.org/3/library/inspect.html#inspect.iscoroutinefunction) instead of the one from asyncio

Reply:
@bellini666 I find it difficult to explain this situation easily and clearly. The official document states, "inspect.iscoroutinefunction(object) Changed in version 3.12: Sync functions marked with markcoroutinefunction() now return True." I am considering noting this as a comment verbatim. How about this idea?